nixos/modules/system/etc/setup-etc.pl

```diff
@@ -1,82 +1,85 @@
+#! @perl@/bin/perl
+
+use feature qw(signatures);
 use strict;
-use File::Find;
-use File::Copy;
-use File::Path;
-use File::Basename;
-use File::Slurp;
+use warnings;
+
+use File::Basename qw(dirname);
+use File::Copy qw(copy);
+use File::Find qw(find);
+use File::Path qw(make_path rmtree);
+use File::Slurp qw(read_file write_file);
+use File::Spec::Functions qw(catfile);
 
-my $etc = $ARGV[0] or die;
+my $etc = $ARGV[0] or die("This script must be called with an argument which expresses the new /etc directory in the nix store");
 my $static = "/etc/static";
+my @errors;
 
-sub atomicSymlink {
-    my ($source, $target) = @_;
+# Create a symlink atomically from argument 2 -> argument 1
+sub atomic_symlink ($source, $target) {
     my $tmp = "$target.tmp";
-    unlink $tmp;
-    symlink $source, $tmp or return 0;
-    rename $tmp, $target or return 0;
+    unlink($tmp);
+    symlink($source, $tmp) or return 0;
+    rename($tmp, $target) or return 0;
```

Comment:
```suggestion
    symlink($source, $tmp) or return;
    rename($tmp, $target) or return;
```

`return;` is preferred to `return 0;`, for the same reason `return undef;` is discouraged

https://metacpan.org/pod/Perl::Critic::Policy::Subroutines::ProhibitExplicitReturnUndef